_posts/2018-11-16-scalafix-scalameta.md

```diff
@@ -0,0 +1,136 @@
+---
+category: blog
+by: Ólafur Páll Geirsson, Eugene Burmako
+title: "Build code analyzers with Scalameta and Scalafix"
+---
+
+We are excited to announce the release of
+[Scalameta v4.0.0](https://github.com/scalameta/scalameta/releases/tag/v4.0.0)
+and
+[Scalafix v0.9.0](https://github.com/scalacenter/scalafix/releases/tag/v0.8.0-RC1)
+introducing new APIs enabling more advanced source code analysis. Scalameta is a
+library to read, analyze, transform and generate Scala programs. Scalafix is a
+refactoring and linting tool.
+
+This release is the result of a close collaboration between the Scala Center and
+Twitter along with contributions from over 30 contributors in the community. Big
+thanks to everybody who made this happen!
+
+## Get started with the new documentation
+
+A big focus of this release has been documentation. Scalameta and Scalafix have
+new websites with guides, cookbooks and reference documents.
+
+Visit [scalameta.org](https://scalameta.org/) to see the new Scalameta
+documentation. Key pages of the site include:
+
+- [SemanticDB specification](https://scalameta.org/docs/semanticdb/specification.html):
+  learn the details of the new semantic API.
+- [SemanticDB guide](https://scalameta.org/docs/semanticdb/specification.html):
+  get started with the SemanticDB command-line tools `metac`, `metacp` and
+  `metap`.
+- [Trees guide](https://scalameta.org/docs/trees/guide.html): learn how to
+  parse, construct, traverse and transform Scala syntax trees.
+- [ScalaFiddle playground](https://scalameta.org/docs/trees/scalafiddle.html):
+  explore Scalameta APIs directly in the browser with access to code completions
+  and fast edit/compile/run feedback. Big thanks to
+  [scalafiddle.io](https://scalafiddle.io/) for providing a great service!
+
+Visit
+[scalacenter.scalafix.io/scalafix](https://scalacenter.scalafix.io/scalafix/) to
+see the new Scalafix documentation. Key pages of the site include:
+
+- [Implementing custom rules tutorial](https://scalacenter.github.io/scalafix/docs/developers/tutorial.html):
+  learn all the steps of implementing a custom rewrite or linter rule, starting
+  from tests and ending with publishing.
+- [Patch reference](https://scalacenter.github.io/scalafix/docs/developers/tutorial.html):
```

Comment:
Should probably be: https://scalacenter.github.io/scalafix/docs/developers/patch.html